server/lib/transactions.ts

```diff
@@ -57,140 +56,140 @@ export function getTransactions(collectiveids, startDate = new Date('2015-01-01'
+    fxRates.expenseToCollective = expenseToHostFxRate;
+  } else if (expense.currency === expense.collective.currency) {
+    // Or the expense has the same currency as its collective and we need to convert it to the host currency
+    fxRates.collectiveToHost = expenseToHostFxRate;
+    fxRates.expenseToCollective = 1;
   } else {
-    // If manual (add funds or manual reimbursement of an expense)
-    hostCurrencyFxRate = await getFxRate(expense.currency, host.currency, expense.incurredAt || expense.createdAt);
-    paymentProcessorFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * paymentProcessorFeeInHostCurrency);
-    hostFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * hostFeeInHostCurrency);
-    platformFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * platformFeeInHostCurrency);
+    // Or we're in a tricky situation where we have neither the currency of the host or the currency of the collective
+    throw new Error(
+      'Multi-currency expenses are not supported for collectives that have a different currency than their hosts',
+    );
   }
 
-  // We assume that all expenses are in Collective currency
-  // (otherwise, ledger breaks with a triple currency conversion)
+  // Compute the amounts in the proper currency
+  return {
+    fxRates,
+    amount: {
+      inHostCurrency: Math.round(expense.amount * fxRates.expenseToHost),
+      inCollectiveCurrency: Math.round(expense.amount * fxRates.expenseToCollective),
+      inExpenseCurrency: expense.amount,
+    },
+    paymentProcessorFee: {
+      inHostCurrency: fees.paymentProcessorFeeInHostCurrency,
+      inCollectiveCurrency: Math.round(fees.paymentProcessorFeeInHostCurrency / fxRates.collectiveToHost),
+      inExpenseCurrency: Math.round(fees.paymentProcessorFeeInHostCurrency / fxRates.expenseToHost),
+    },
+    hostFee: {
+      inHostCurrency: fees.hostFeeInHostCurrency,
+      inCollectiveCurrency: Math.round(fees.hostFeeInHostCurrency / fxRates.collectiveToHost),
+      inExpenseCurrency: Math.round(fees.hostFeeInHostCurrency / fxRates.expenseToHost),
+    },
+    platformFee: {
+      inHostCurrency: fees.platformFeeInHostCurrency,
+      inCollectiveCurrency: Math.round(fees.platformFeeInHostCurrency / fxRates.collectiveToHost),
+      inExpenseCurrency: Math.round(fees.platformFeeInHostCurrency / fxRates.expenseToHost),
+    },
+  };
+};
+
+/**
+ * A function to create transactions for a given expense that is agnostic of the payout method
+ *
+ * TODO: This function should accept an `amount` to automatically represent what was really paid from the payment provider,
+ * in case it differs from expense.amount (e.g. when fees are put on the payee)
+ */
+export async function createTransactionsFromPaidExpense(
+  host,
+  expense,
+  fees: FEES_IN_HOST_CURRENCY = DEFAULT_FEES,
+  /** Set this to a different value if the expense was paid in a currency that differs form the host's */
+  expenseToHostFxRateConfig: number | 'auto',
```

Comment:
Do we need this `auto` ? We could just do,

```suggestion
  expenseToHostFxRateConfig: number = null,
```

And then set, 

```js
// Get the right FX rate for the expense
  const expenseToHostFxRate =
    expenseToHostFxRateConfig || await getFxRate(expense.currency, host.currency, expense.incurredAt || expense.createdAt);
```

This way we can get rid of the duplicating `auto` arg in the function references.

Reply:
I want the automatic FX rate to be an explicit feature that developers enable consciously. If we make it optional, anyone implementing a new payout provider could end up calling this function with the simplest prototype, while we usually want to retrieve the FX rate from the payout provider. Only expenses paid manually or using our core system are supposed to have an automatic fx rate.